ide/project.dependency/src/org/netbeans/modules/project/dependency/ArtifactSpec.java

```diff
@@ -75,17 +86,45 @@ public enum VersionKind {
         this.optional = optional;
         this.data = impl;
         this.type = type;
+        this.location = location;
         this.localFile = localFile;
     }
 
     public T getData() {
         return data;
     }
 
+    /**
+     * Returns local file which the artifact represents. For library (dependencies) artifacts,
+     * the file is the library consumed, e.g. from a local repository. For outputs, the artifact
+     * represents the build output, usually in project's build directory. Note that FileObject for 
+     * a dependency and its corresponding Project may not be the same.
+     * 
```

Comment:
Probably there should be mentioned, that can return null, if the local file doesn't exist?